src/GitLabGitProvider/src/GitLabProvider/Client/GitLabClient.cs

```diff
@@ -172,6 +172,42 @@ public async Task<bool> CreateBranch(string branchname, string repositoryId)
             return null;
         }
 
+        public async Task<List<GitLabTreeFile>> GetFiles(string repositoryId, string branchName)
+        {
+            var projectId = repositoryId;
+            var pageNumber = "1";
+            var allFiles = new List<GitLabTreeFile>();
+            try
+            {
+                
+                while (!string.IsNullOrWhiteSpace(pageNumber))
+                {
+                    var request = new HttpRequestMessage(HttpMethod.Get, $"{_baseUrl}/api/v4/projects/{projectId}/repository/tree?recursive=true&ref={branchName}&per_page=10&page={pageNumber}");
+                    request.Headers.Authorization = new AuthenticationHeaderValue("Bearer", _accesstoken);

```

Comment:
kommer vi behöva skicka in en till secret för accesstoken till gitlab?

Reply:
Det gör vi redan.